src/main/scala/za/co/absa/spark/commons/version/SparkVersionGuard.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spark.commons.version
+
+import org.slf4j.{Logger, LoggerFactory}
+import za.co.absa.commons.version.Version
+import za.co.absa.commons.version.Version.VersionStringInterpolator
+import za.co.absa.commons.version.impl.SemVer20Impl.SemanticVersion
+
+object SparkVersionGuard {
+  val minSpark2XVersionIncluded: SemanticVersion = semver"2.4.2"
+  val maxSpark2XVersionExcluded: SemanticVersion = semver"2.4.8"
+
+  val minSpark3XVersionIncluded: SemanticVersion = semver"3.0.0"
+  val maxSpark3XVersionExcluded: SemanticVersion = semver"3.2.0"
+
+  /**
+   * Populates the version guard with the defaults
+   */
+  def fromDefaultSparkCompatibilitySettings: SparkVersionGuard =
+    SparkVersionGuard(minSpark2XVersionIncluded, Some(maxSpark3XVersionExcluded))
+
+  /**
+   * Populates the version guard with the defaults for Spark 2.X
+   */
+  def fromSpark2XCompatibilitySettings: SparkVersionGuard =
+    SparkVersionGuard(minSpark2XVersionIncluded, Some(maxSpark2XVersionExcluded))
```

Comment:
I do not think this will be doing what we want. If they release 2.4.9 we need to update the version. Instead of making it [2.4.2, 2.4.8], make it [2.4.2, 3.0.0)

[ or ] being a range that is inclusive 
( or ) being a range that is exclusive

Reply:
For version 3 it would be [3.0.0, 4.0.0)?